Compile failed on MacOS 12.4 (arm chip)

clang 13 or gcc 12 both compile error 
```
./cpp-flex.h:9:69: error: expected expression
  "m4_define([[M4_PROPERTY_TRACE_LINE_REGEXP]], [[^#line ([0-9]+) ",\"(.*)\"]])
                                                                    ^
./cpp-flex.h:9:70: warning: missing terminating '"' character [-Winvalid-pp-token]
  "m4_define([[M4_PROPERTY_TRACE_LINE_REGEXP]], [[^#line ([0-9]+) ",\"(.*)\"]])

```

It look like m4 process error.

My m4 version is 

<img width="533" alt="image" src="https://user-images.githubusercontent.com/13445879/183837348-d5b07ba1-0795-46d3-97d2-22479f50e845.png">

and generate `cpp-flex.h`

![image](https://user-images.githubusercontent.com/13445879/183837463-3b84ede5-9a99-4d02-af10-fd385af7f74f.png)

As I'm not familiar with m4, why it generate the ",\" ?

--- a/src/main.c
+++ b/src/main.c
@@ -295,7 +295,7 @@ void initialize_output_filters(void)
 	if ( !(m4 = getenv("M4"))) {
 		m4 = M4;
 	}
-	filter_create_ext(output_chain, m4, "-P", 0);
+	filter_create_ext(output_chain, m4, "-P", (char *) 0);
 	filter_create_int(output_chain, filter_fix_linedirs, NULL);
 
 	/* For debugging, only run the requested number of filters. */

